dask_geomodeling/raster/elemwise.py

```diff
@@ -746,3 +749,81 @@ def process(process_kwargs, *args):
             values[index] = data[index]
 
         return {"values": values, "no_data_value": fillvalue}
+
+
+class BaseLogExp(BaseSingle):
+    """
+    Base Block for Exp, Log and Log10 Blocks.
+    """
+    def __init__(self, x):
+        if x.dtype == np.dtype("bool"):
+            raise TypeError("input block must not have boolean dtype")
+        super(BaseLogExp, self).__init__(x)
+
+    def get_sources_and_requests(self, **request):
+        process_kwargs = {"dtype": self.dtype.name, "fillvalue": self.fillvalue}
+        return [(process_kwargs, None), (self.args[0], request)]
+
+    @property
+    def dtype(self):
+        # use at least float32
+        return np.result_type(np.float32, *self.args)
+
+    @property
+    def fillvalue(self):
+        return get_dtype_max(self.dtype)
+
+
+class Exp(BaseLogExp):
+    """
+    Return e raised to the power of the raster values.
```

Comment:
Good to note that out-of-range values (like Exp(1000000)) will become nodata as well

Reply:
Except yet it doesn't ;-) - do we have this kind of protections for other blocks as well? Coming to think of it, what do you think of putting this non-finite masking in the `wrap_math_process_func`? It took my a while to find out that the MemorySource and the MockRaster already do this.

And then, do you prefer out-of-range values on individual block docstrings, or shall we make it a general not, and where should it go then?

Reply:
I like to lift this `isfinite` masking to the `wrap_math_process_func`. Also for Multiply the issue will be possible definitely